server/text_generation_server/models/custom_modeling/qwen2_vl.py

```diff
@@ -424,33 +425,50 @@ def get_position_ids(
             device=batch_input_ids.device,
         )
         d = batch_input_ids.device
-        if image_grid_thw is not None:
-            image_index = 0
+
+        # Handle both image and video tokens
+        if image_grid_thw is not None or video_grid_thw is not None:
+            vision_index = 0
             llm_pos_ids_list = []
 
             for i, input_ids in enumerate(batch_input_ids):
                 vision_start_indices = torch.argwhere(
                     input_ids == self.vision_start_token_id
                 ).squeeze(1)
                 vision_tokens = input_ids[vision_start_indices + 1]
-                # only copy the sum of the image tokens GPU<->CPU
+
+                # only copy the sum of the image and video tokens GPU<->CPU
                 image_count = (vision_tokens == self.image_token_id).sum().item()
+                video_count = (vision_tokens == self.video_token_id).sum().item()
 
                 current_pos = 0
-                for _ in range(image_count):
-                    # copy the value position of the next image token from GPU<->CPU
-                    next_image_pos = (
-                        (input_ids[current_pos:] == self.image_token_id)
+                for _ in range(image_count + video_count):
```

Comment:
This looks pretty bloated.

It was like that before but it's still seems quite bloated.
There should be 1 replace for all images, 1 replace for all videos.

Not super urgent if this works (as it's limited to qwen).

Also seems everything named image was renamed to video.. Sounds like better names can be found.